assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useUploader } from '../../uploader';
+import { useAPI } from '../../api';
+import {
+  getResourceFromLocalFile,
+  getResourceFromUploadAPI,
+  getResourceFromAttachment,
+} from '../../../app/media/utils';
+
+function useUploadMedia({
+  media,
+  pagingNum,
+  searchTerm,
+  fetchMediaStart,
+  setMedia,
+  fetchMediaError,
+}) {
+  const { uploadFile } = useUploader();
+  const {
+    actions: { getMedia },
+  } = useAPI();
+
+  const resetMedia = useCallback(
+    ({ pagingNum: p = 1 } = {}) => {
+      fetchMediaStart({ pagingNum: p });
+      getMedia({ searchTerm, pagingNum: p })
+        .then(({ data }) => {
+          setMedia({
+            media: data,
+          });
+        })
+        .catch(fetchMediaError);
+    },
+    [setMedia, fetchMediaError, fetchMediaStart, getMedia, searchTerm]
+  );
+
+  const uploadMediaFromLibrary = useCallback(
```

Comment:
Could we make a generic function here, that can be used library and workspace. There basically do that same thing by apart the insert in canva with workspace. At the very least the code should have reusable part and look similar.

Reply:
How and where was this resolved?